### ISSUE TITLE
State leakage between users is possible using this module

### ISSUE THREAD

**Issue by bcairns** [joshnuss/svelte-persisted-store#306]
We have observed that when used with SvelteKit, this module allows writes on the server during SSR, and this can lead to state leaking between users.

This is due to the module-level `stores` constant, which acts as a cache.

## Minimal test case:

- Create a new SvelteKit project and then install svelte-persisted-store (eg `npx sv create persisted-test`, `npm install svelte-persisted-store`)

- Create the following two files:

`routes/+page.svelte`:

```
<script>
    import { persisted } from "svelte-persisted-store";

    const recentlyViewedId = persisted('RECENTLY_VIEWED_ID', null);
</script>

Recently viewed ID: {$recentlyViewedId}
```

`routes/product/+page.svelte`:

```
<script>
    import { persisted } from "svelte-persisted-store";

    const recentlyViewedId = persisted('RECENTLY_VIEWED_ID', null);

    $recentlyViewedId = 123;
</script>

<h1>Product Page 123</h1>
```

- Run the server (eg `npm run dev`)

- In one browser, visit the /product route (eg http://localhost:5173/product)

- In a completely separate browser, visit the "homepage" (eg http://localhost:5173).  In the **page source** (or with Javascript disabled) observe that the string "Recently viewed ID: 123" is present, containing the value "123" that was written by the other browser during page render.

## Results

Using the library in this fashion allows state (that a developer might reasonably expect to only persist to local storage) to show up for other users.

It can certainly be argued that developers should take care to only write to this store from the client (eg from `onMount()` or an event handler), but others will make this mistake too.

**Comment by Nicicalu** [comment]
@bcairns I thought about using SSR again with this library, but I decided against it because of your comment. Thanks!

**Comment by joshnuss** [comment]
This store is based on `window.localStorage`, which is a browser-only feature.
So it really shouldn't be used server-side.

In the future Svelte 5 version, I hope to raise an error if attempted to run server-side (ie `window` is not defined).

**Comment by webJose** [comment]
Hello, josh.  I think the problem is that internally, this package uses a global map to never create more than one store for the same storage key.  This setup works OK in the client side, but in the server side, this leaks information from one user onto another user, because on the server, singletons hold but are used by the HTTP server for all users.

**Comment by joshnuss** [comment]
Good point @webJose,

I have created a fix #307 which removes the use of global state during SSR.

Wanna take a look?

**Comment by bcairns** [comment]
Thank you very much for addressing this.

Do you have any estimate of when this fix will make it into a versioned release?
